src/vs/platform/progress/common/progress.ts

```diff
@@ -54,6 +54,7 @@ export interface IProgressOptions {
 	readonly title?: string;
 	readonly source?: string | { label: string; id: string };
 	readonly total?: number;
+	readonly type?: 'syncing' | 'loading';
```

Comment:
Adding a new property to `IProgressOptions` means we have to support it in all locations that can show progress. So I would rather have a progress options for the `IProgressWindowOptions` which is the status bar if possible?

Reply:
I haven't used progress service yet so this might be a silly question. To do that, should this line:

https://github.com/microsoft/vscode/blob/e4c385658bf17265730e3fde210266dbdfcaa971/src/vs/workbench/services/progress/browser/progressService.ts#L48

Use `options: IProgressOptions | IProgressWindowOptions`? Or `options: IProgressOptions | IProgressOptions & IProgressWindowOptions`?

Reply:
https://github.com/microsoft/vscode/pull/155616/commits/9d20460909ec0bd77b0e85ec70b75a847597f4a5